tests/api/test_meta.py

```diff
@@ -0,0 +1,4421 @@
+    # given
+    variables = {"id": graphene.Node.to_global_id("Category", category.pk)}
+
+    # when
+    response = api_client.post_graphql(QUERY_CATEGORY_PRIVATE_META, variables)
+
+    # then
+    assert_no_permission(response)
+
+
+def test_query_private_meta_for_category_as_customer(user_api_client, category):
+    # given
+    variables = {"id": graphene.Node.to_global_id("Category", category.pk)}
+
+    # when
+    response = user_api_client.post_graphql(QUERY_CATEGORY_PRIVATE_META, variables)
+
+    # then
+    assert_no_permission(response)
+
+
+def test_query_private_meta_for_category_as_staff(
+    staff_api_client, category, permission_manage_products
+):
+    # given
+    category.store_private_meta({PRIVATE_KEY: PRIVATE_VALUE})
+    category.save(update_fields=["private_meta"])
+    variables = {"id": graphene.Node.to_global_id("Category", category.pk)}
+
+    # when
+    response = staff_api_client.post_graphql(
+        QUERY_CATEGORY_PRIVATE_META,
+        variables,
+        [permission_manage_products],
+        check_no_permissions=False,
+    )
+    content = get_graphql_content(response)
+
+    # then
+    metadata = content["data"]["category"]["privateMetadata"][0]
+    assert metadata["key"] == PRIVATE_KEY
+    assert metadata["value"] == PRIVATE_VALUE
+
+
+def test_query_private_meta_for_category_as_service_account(
+    service_account_api_client, category, permission_manage_products
+):
+    # given
+    category.store_private_meta({PRIVATE_KEY: PRIVATE_VALUE})
+    category.save(update_fields=["private_meta"])
+    variables = {"id": graphene.Node.to_global_id("Category", category.pk)}
+
+    # when
+    response = service_account_api_client.post_graphql(
+        QUERY_CATEGORY_PRIVATE_META,
+        variables,
+        [permission_manage_products],
+        check_no_permissions=False,
+    )
+    content = get_graphql_content(response)
+
+    # then
+    metadata = content["data"]["category"]["privateMetadata"][0]
+    assert metadata["key"] == PRIVATE_KEY
+    assert metadata["value"] == PRIVATE_VALUE
+
+
+QUERY_COLLECTION_PRIVATE_META = """
+    query collectionMeta($id: ID!){
+        collection(id: $id){
+            privateMetadata{
+                key
+                value
+            }
+        }
+    }
+"""
+
+
+def test_query_private_meta_for_collection_as_anonymous_user(api_client, collection):
+    # given
+    variables = {"id": graphene.Node.to_global_id("Collection", collection.pk)}
+
+    # when
+    response = api_client.post_graphql(QUERY_COLLECTION_PRIVATE_META, variables)
+
+    # then
+    assert_no_permission(response)
+
+
+def test_query_private_meta_for_collection_as_customer(user_api_client, collection):
+    # given
+    variables = {"id": graphene.Node.to_global_id("Collection", collection.pk)}
+
+    # when
+    response = user_api_client.post_graphql(QUERY_COLLECTION_PRIVATE_META, variables)
+
+    # then
+    assert_no_permission(respons
```

Comment:
Great work @fowczarek !
I am really impressed how many tests you added but we should split it to separate files. Searching a single test in such a huge file can be really hard. 
I would suggest splitting them to `api/meta/test_(private)_meta` or `api/test_(private)_meta`  or (public/private) and per type (mutation/query) == `api/meta/test_piravte_meta_mutations`